internal/common/proto/protoutil_test.go

```diff
@@ -0,0 +1,65 @@
+package protoutil
+
+import (
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/armadaproject/armada/internal/common/compress"
+	"github.com/armadaproject/armada/pkg/armadaevents"
+)
+
+var (
+	msg              = &armadaevents.CancelJob{JobId: armadaevents.ProtoUuidFromUuid(uuid.New())}
+	compressor       = compress.NewThreadSafeZlibCompressor(1024)
+	decompressor     = compress.NewThreadSafeZlibDecompressor()
+	marshalledMsg, _ = proto.Marshal(msg)
+	compressedMsg, _ = compressor.Compress(marshalledMsg)
+	invalidMsg       = []byte{0x3}
+)
+
+func TestUnmarshall_Valid(t *testing.T) {
```

Comment:
One test pattern I've started following is to use table driven tests.  

https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
I do that but I'm not sure it's useful here as we only have a couple of tests

Reply:
Yea I see that.  No worries then.